### DISCUSSION TITLE
Tampermonkey Source State

### DISCUSSION THREAD

**Discussion by Malix-Labs** [Tampermonkey/tampermonkey#1735] [Q&A] [answered]
Hello!

I noticed the code in this current repository seems to not match what the current Tampermonkey product is.

Does it became closed-source?

**Comment by 7nik** [comment]
> [README.md](https://github.com/Tampermonkey/tampermonkey#readme)
This repository contains the source of the Tampermonkey extension up to version 2.9. All newer versions are distributed under a proprietary license.

Yes, it has been closed-sourced since version 3.0

**Comment by derjanb** [comment, marked as answer]
> Does it became closed-source?

As the developer of Tampermonkey, I have always worked on it in a private repository.  The reason for this is that I enjoy working alone without having to consider external contributions or feedback regarding the implementation and architecture for example.  While I did occasionally synchronize the GitHub repository with my private repository between versions 1.0.0.1 and 2.9.0 from 2010 to 2013, the syncing was sporadic and sometimes occurred with a two-year gap.

However, the Mozilla Add-ons and Opera teams are receiving a source package to build and review Tampermonkey with every new version, and when I worked with the Microsoft Edge team on extension support for the first version of the Edge browser, they also had full access to the source code. Additionally, Firefox recently made Tampermonkey available for their Android browser after a review.

**Reply by Malix-Labs** [reply]
That's nice to hear.

Do you consider opening it in the future?
If people crying about your code, GitHub has lot of features to offer to shut their mouth up :))

**Reply by stokito** [reply]
I switched to ViolentMonkey. It's always better to use FLOSS even if it lacks of features.

**Reply by Malix-Labs** [reply]
@stokito 
> I switched to ViolentMonkey. It's always better to use FLOSS even if it lacks of features.

I tend to say that for dev products such as BaaS, OSS is a requirement; but for end-user products, I understand close-sourcing it. Tampermonkey lays in between end-user and dev products, and so is its sourcing system.

I would love tampermonkey being OSS tho, as it has the potential to be pushed by it while also being free to use, so there wouldn't be any downside other than open feedback, and as @derjanb said:
> I enjoy working alone without having to consider external contributions or feedback regarding the implementation and architecture for example.

So I understand if he just doesn't care about open-sourcing it.

Still, @derjanb, if open feedbacks are the only thing blocking you from OSS it, I will be glad to be your feedback gatekeeper as a triage guy.

**Reply by stokito** [reply]
You always can just ignore external contributions but keep sources open. If something critical then someone will fork it. Right now the ViolentMonkey is just better. The only one reason why I had to use again the Tampermonkey is because it's the only one extension that is supported by Firefox Mobile.
And it's not so easy to do for them https://github.com/violentmonkey/violentmonkey/issues/1042

So while author tried to keep focused and not spent own time for PRs instead many other peoples had to spent their own time to write a FOSS alternative. And eventually most if not all users will switch to the VM.
So the Tampermoneky become abandoned and all the author's time will be just lost

**Reply by merely-present** [reply]
This was disappointing to find. I had thought that this was open source, and even just donated casually thinking "ah, I use this a lot and its open source!"

Its particularly disappointing because I wanted to add the ability to require other installed scripts, but free is free I suppose. the oss doesn't have to part of it.

I would honestly be ok with the thrown over the wall approach where other peoples prs weren't looked into if I could just see/modify my own versions. Of course that's all being said with no sense of entitlement. We signed no agreement. I am owed nothing. I am just expressing an opinion on this discussion page.

**Reply by stokito** [reply]
@CS-Account try ViolentMonkey

**Reply by Malix-Labs** [reply]
@CS-Account I agree

**Comment by bn-l** [comment]
In my opinion this is kind of fraudulent not making this very clear in the readme (I think a majority of people coming here to see if it's open source are not looking at the last commit time). 

I would like to see something like this:

# NOTICE: Tampermonkey is closed source

**Reply by KevinBlue28** [reply]
It actually does say so in the README. The [commit](https://github.com/Tampermonkey/tampermonkey/commit/e547f538f49a27174568c1b19839a73bd5f8fb10) about the license change was created March 24, 2016. I do agree that bold should have been used instead of italics for stronger emphasis.
